### PR TITLE
[Solved #84]  create start-race controller

### PR DESCRIPTION
**Pull request by supunTE**
Solved #84 
- Create util for generating race text
- Client mock update (create a function to emit start race)
- Enhance type declaration in logger.
- Update start countdown protocol docs
- **Create start-race controller**
- Implement validations to check player's tournament using the socket room

### CODE REVIEW

File: apps/server/src/utils/generateRaceText.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import axios from 'axios';
+
+export const generateRaceText = async (): Promise<string> => {
```
Comment (dalanad) [inline apps/server/src/utils/generateRaceText.ts:3]: shouldn't this be parameterized to include paragraph count and other attributes? of text

File: apps/server/src/controllers/start-race.controller.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  tournamentId: AppTournamentId;
+}
+
+const logger = new Logger('start-race.controller');
+
+export const startRaceController = async ({
+  context,
+  playerId,
+  tournamentId,
+}: StartRaceRequestArgs): Promise<void> => {
```
Comment (dalanad) [inline apps/server/src/controllers/start-race.controller.ts:25]: As we have already discussed shall we send  error messages back to the user

**Review by dalanad — CHANGES_REQUESTED** [review]

File: apps/server/src/utils/generateRaceText.ts (hunk 1 above)
Comment (supunTE) [reply apps/server/src/utils/generateRaceText.ts:3]: Currently, we only need 1 paragraph for race. we can expand it later.

File: apps/server/src/controllers/start-race.controller.ts (hunk 2 above)
Comment (supunTE) [reply apps/server/src/controllers/start-race.controller.ts:25]: Opened a separate issue for this https://github.com/the-ai-team/razor/issues/80

**Review by dalanad — APPROVED** [review]

**Pull request merged**